### PR TITLE
Additional Features to Heimdall

### PR DESCRIPTION
**Pull request by taylorhakes**
As my linked ticket suggested https://github.com/fterh/heimdall/issues/47. Since this is likely a breaking change, I would recommend we increase the major version. We could keep backwards compatibility by using the body or something, but I feel that is not ideal. Let me know your thoughts on these changes.

**Features Added**
- Adds custom alias names as email subject
- Support for outbound aliases in email subject
- Use encrypted email password in SSM

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,6 +4,25 @@
 [![Maintainability](https://api.codeclimate.com/v1/badges/8400b327ea3d328c9f5e/maintainability)](https://codeclimate.com/github/fterh/heimdall/maintainability)
 [![Test Coverage](https://api.codeclimate.com/v1/badges/8400b327ea3d328c9f5e/test_coverage)](https://codeclimate.com/github/fterh/heimdall/test_coverage)
 
+Features added:
```
Comment (taylorhakes) [inline README.md:7]: This is a remnants of how my ReadME changed. I can update the readme correctly if you would like to move forward.

File: serverless.yml [hunk 2]
```diff
@@ -2,24 +2,35 @@ service: heimdall
...
+        - ssm:GetParameter
+      Resource:
+        Fn::Join:
+          - ''
+          -
+            - 'arn:aws:ssm:'
+            - Ref: AWS::Region
+            - ':'
+            - Ref: AWS::AccountId
+            - ':parameter/wabble_smtp_password'
```
Comment (taylorhakes) [inline serverless.yml:32]: I need to change this to a parameter

**Comment by fterh** [comment]
Thanks @taylorhakes! I'm primarily interested in only the "Support for outbound aliases in email subject" feature, and I'll look through the code in more detail when I have time. From a quick scan, it seems that the current implementation associates each alias with an outbound email address. My use case requires specifying the outbound address at send-time not generate-time, because I won't know that ahead of time.

I'm going to close this request for now (sorry!) but thank you for the ideas!

**Comment by taylorhakes** [comment]
No problem. Completely up to you.

> My use case requires specifying the outbound address at send-time not generate-time, because I won't know that ahead of time.

I thought about doing that originally, but decided against it because it became really awkward. Where do I put the outgoing email address? In the subject? In the body? It just felt wrong and error prone for users either way. I decided instead to create at generate time. If you come up with a simple way of specifying it at send-time, I would be very interested to learn.

**Pull request closed without merging**